source/common/thread/terminate_thread.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Thread {
+/**
+ * Tries to terminates the process by killing the thread specified by
+ * the ThreadId. The implementation is platform dependent and currently
+ * only works on platforms that support SIGABRT.
+ *
+ * Returns the result from the platform specific function (i.e. kill) to terminate
+ * the thread. If the platform is currently unsupported, this will return false.
```

Comment:
This does not return the result of the platform specific function, it returns true if the platform specific function succeeded.  See implementation, return is: kill() == 0;

Reply:
good catch I've made it: 

Returns true if the platform specific function to terminate the thread succeeded (i.e. kill() == 0). If the platform is currently unsupported, this will return false.